### ISSUE TITLE
Brew install opencv fails at 99% 

### ISSUE THREAD

**Issue by kscasado** [Homebrew/homebrew-science#2905]
I have tried --build-from-source as well with no luck 
have not seen any other issues that are similar
also have brewed python
Below are the last few lines:

[ 99%] Built target opencv_test_contrib
[ 99%] Building CXX object modules/gpu/CMakeFiles/opencv_perf_gpu.dir/perf/perf_denoising.cpp.o
cd /tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/macbuild/modules/gpu && /home/kcasado/.linuxbrew/bin/g++-5    -Os -w -pipe -march=core2    -fsigned-char -W -Wall -Werror=return-type -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-array-bounds -Wno-aggressive-loop-optimizations -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -msse -msse2 -msse3 -ffunction-sections   -DNDEBUG -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/gpu/perf -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/video/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/calib3d/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/features2d/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/highgui/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/imgproc/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/flann/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/core/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/ts/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/gpu/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/photo/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/objdetect/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/legacy/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/ml/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/macbuild/modules/gpu -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/gpu/src -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/gpu/test -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/gpu/src/cuda -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/macbuild -isystem /home/kcasado/.linuxbrew/include/eigen3     -Winvalid-pch -o CMakeFiles/opencv_perf_gpu.dir/perf/perf_denoising.cpp.o -c /tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/gpu/perf/perf_denoising.cpp
[ 99%] Building CXX object modules/gpu/CMakeFiles/opencv_perf_gpu.dir/perf/perf_labeling.cpp.o
cd /tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/macbuild/modules/gpu && /home/kcasado/.linuxbrew/bin/g++-5    -Os -w -pipe -march=core2    -fsigned-char -W -Wall -Werror=return-type -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-array-bounds -Wno-aggressive-loop-optimizations -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -msse -msse2 -msse3 -ffunction-sections   -DNDEBUG -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/gpu/perf -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/video/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/calib3d/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/features2d/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/highgui/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/imgproc/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/flann/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/core/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/ts/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/gpu/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/photo/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/objdetect/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/legacy/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/ml/include -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/macbuild/modules/gpu -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/gpu/src -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/gpu/test -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/gpu/src/cuda -I/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/macbuild -isystem /home/kcasado/.linuxbrew/include/eigen3     -Winvalid-pch -o CMakeFiles/opencv_perf_gpu.dir/perf/perf_labeling.cpp.o -c /tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/modules/gpu/perf/perf_labeling.cpp
[ 99%] Linking CXX executable ../../bin/opencv_perf_gpu
cd /tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/macbuild/modules/gpu && /home/kcasado/.linuxbrew/Cellar/cmake/3.3.2/bin/cmake -E cmake_link_script CMakeFiles/opencv_perf_gpu.dir/link.txt --verbose=1
/home/kcasado/.linuxbrew/bin/g++-5   -Os -w -pipe -march=core2    -fsigned-char -W -Wall -Werror=return-type -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-array-bounds -Wno-aggressive-loop-optimizations -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -msse -msse2 -msse3 -ffunction-sections   -DNDEBUG   -L/home/kcasado/.linuxbrew/lib -Wl,-rpath,/home/kcasado/.linuxbrew/lib    CMakeFiles/opencv_perf_gpu.dir/perf/perf_objdetect.cpp.o CMakeFiles/opencv_perf_gpu.dir/perf/perf_calib3d.cpp.o CMakeFiles/opencv_perf_gpu.dir/perf/perf_features2d.cpp.o CMakeFiles/opencv_perf_gpu.dir/perf/perf_matop.cpp.o CMakeFiles/opencv_perf_gpu.dir/perf/perf_core.cpp.o CMakeFiles/opencv_perf_gpu.dir/perf/perf_main.cpp.o CMakeFiles/opencv_perf_gpu.dir/perf/perf_video.cpp.o CMakeFiles/opencv_perf_gpu.dir/perf/perf_imgproc.cpp.o CMakeFiles/opencv_perf_gpu.dir/perf/perf_filters.cpp.o CMakeFiles/opencv_perf_gpu.dir/perf/perf_denoising.cpp.o CMakeFiles/opencv_perf_gpu.dir/perf/perf_labeling.cpp.o  -o ../../bin/opencv_perf_gpu -rdynamic ../../lib/libopencv_gpu.so.2.4.12 ../../lib/libopencv_ts.a -ldl -lm -lpthread -lrt ../../lib/libopencv_legacy.so.2.4.12 ../../lib/libopencv_ml.so.2.4.12 ../../lib/libopencv_objdetect.so.2.4.12 ../../lib/libopencv_photo.so.2.4.12 ../../lib/libopencv_calib3d.so.2.4.12 ../../lib/libopencv_features2d.so.2.4.12 ../../lib/libopencv_flann.so.2.4.12 ../../lib/libopencv_highgui.so.2.4.12 ../../lib/libopencv_video.so.2.4.12 ../../lib/libopencv_imgproc.so.2.4.12 ../../lib/libopencv_core.so.2.4.12 -Wl,-rpath,/home/kcasado/.linuxbrew/Cellar/opencv/2.4.12/lib 
make[2]: Leaving directory `/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/macbuild'
[ 99%] Built target opencv_perf_gpu
make[2]: Leaving directory`/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/macbuild'
make[1]: **\* [modules/python/CMakeFiles/opencv_python.dir/all] Error 2
make[1]: Leaving directory `/tmp/opencv20151016-10625-1xf09vo/opencv-2.4.12/macbuild'
make: **\* [all] Error 2

**Comment by tdsmith** [comment]
please read the troubleshooting guide and link the output of `brew gist-logs opencv`; thanks!

**Comment by kscasado** [comment]
I have done brew doctor and upgrade with no issues
Below is the gist log:
https://gist.github.com/anonymous/bf3fc17a01daaf52b6ea

**Comment by tdsmith** [comment]
It's misdetecting python:
```
make[2]: Entering directory `/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/macbuild'
make[2]: *** No rule to make target `/libpython2.7.so', needed by `lib/cv2.so'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 99%] Building CXX object modules/python/CMakeFiles/opencv_python.dir/src2/cv2.cpp.o
cd /tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/macbuild/modules/python && /home/kcasado/.linuxbrew/bin/g++-5   -DPYTHON_USE_NUMPY=1 -Dopencv_python_EXPORTS -Os -w -pipe -march=core2    -fsigned-char -W -Wall -Werror=return-type -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-array-bounds -Wno-aggressive-loop-optimizations -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -msse -msse2 -msse3 -ffunction-sections -Wno-unused-function   -DNDEBUG -fPIC -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/python/src2 -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/contrib/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/nonfree/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/gpu/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/photo/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/objdetect/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/legacy/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/video/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/ml/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/calib3d/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/features2d/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/highgui/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/imgproc/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/flann/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/core/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/macbuild/modules/python -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/python/src -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/python/include -I/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/macbuild -isystem /home/kcasado/.linuxbrew/include/eigen3 -isystem /home/kcasado/.linuxbrew/Cellar/python/2.7.10_2/include/python2.7 -isystem /home/kcasado/.linuxbrew/lib/python2.7/site-packages/numpy/core/include    -o CMakeFiles/opencv_python.dir/src2/cv2.cpp.o -c /tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/modules/python/src2/cv2.cpp
[ 99%] Linking CXX executable ../../bin/opencv_test_contrib
cd /tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/macbuild/modules/contrib && /home/kcasado/.linuxbrew/Cellar/cmake/3.3.2/bin/cmake -E cmake_link_script CMakeFiles/opencv_test_contrib.dir/link.txt --verbose=1
/home/kcasado/.linuxbrew/bin/g++-5   -Os -w -pipe -march=core2    -fsigned-char -W -Wall -Werror=return-type -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-array-bounds -Wno-aggressive-loop-optimizations -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -msse -msse2 -msse3 -ffunction-sections   -DNDEBUG   -L/home/kcasado/.linuxbrew/lib -Wl,-rpath,/home/kcasado/.linuxbrew/lib    CMakeFiles/opencv_test_contrib.dir/test/test_main.cpp.o  -o ../../bin/opencv_test_contrib -rdynamic ../../lib/libopencv_contrib.so.2.4.12 ../../lib/libopencv_ts.a -ldl -lm -lpthread -lrt ../../lib/libopencv_nonfree.so.2.4.12 ../../lib/libopencv_gpu.so.2.4.12 ../../lib/libopencv_legacy.so.2.4.12 ../../lib/libopencv_ml.so.2.4.12 ../../lib/libopencv_objdetect.so.2.4.12 ../../lib/libopencv_photo.so.2.4.12 ../../lib/libopencv_calib3d.so.2.4.12 ../../lib/libopencv_features2d.so.2.4.12 ../../lib/libopencv_flann.so.2.4.12 ../../lib/libopencv_highgui.so.2.4.12 ../../lib/libopencv_video.so.2.4.12 ../../lib/libopencv_imgproc.so.2.4.12 ../../lib/libopencv_core.so.2.4.12 -Wl,-rpath,/home/kcasado/.linuxbrew/Cellar/opencv/2.4.12/lib 
make[2]: Leaving directory `/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/macbuild'
[ 99%] Built target opencv_test_contrib
make[2]: Leaving directory `/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/macbuild'
make[1]: *** [modules/python/CMakeFiles/opencv_python.dir/all] Error 2
make[1]: Leaving directory `/tmp/opencv20151016-2804-15ezo73/opencv-2.4.12/macbuild'
make: *** [all] Error 2
```

We're passing in `-DPYTHON_LIBRARY=/libpython2.7.so` which is wrong.

**Comment by tdsmith** [comment]
What's the output you get from `python-config --prefix` and `python-config --configdir`?

**Comment by ffmpbgrnn** [comment]
I met the same problem! Any updates?

**Comment by ffmpbgrnn** [comment]
Solved by changing `python-config --configdir` to `python-config --prefix` in opencv.rb.

**Comment by dpo** [comment]
@ffmpbgrnn @kscasado Is this still an issue? It would be easily fixed in a pull request.

**Comment by timotheecour** [comment]
I believe this fixed it : https://github.com/Homebrew/homebrew-science/commit/b0350e62de1deb3ccf20face10e11c1df1bd4820

**Comment by jonchang** [comment]
Assuming this was fixed by #4750, please file a new issue if it wasn't.

**Comment by ffmpbgrnn** [comment]
@jonchang , Yes, it is fixed. Thanks.
